nep-22.mediawiki

```diff
@@ -0,0 +1,73 @@
+<pre>
+  NEP: 22
+  Title: Contract Update Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>, Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2021-04-19
+</pre>
+
+
+==Abstract==
+
+This proposal outlines a standard for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract update.
+
+==Motivation==
+
+As the Neo blockchain scales, Smart Contract Update will become increasingly important. We need a standard for <code>update</code> method in contract to callback native <code>ContractManagement</code> correctly.
+
+==Specification==
+
+All contracts SHOULD implement this NEP in case of any bug hotfix or improvement.
+
+===Methods===
+
+====update====
+
+<pre>
+{
+  "name": "update",
+  "safe": false,
+  "parameters": [
+    {
+      "name": "nefFile",
+      "type": "ByteArray"
+    },
+    {
+      "name": "manifest",
+      "type": "ByteArray"
+    },
+    {
+      "name": "data",
+      "type": "Any"
+    }
+  ],
+  "returntype": "Void"
+}
+</pre>
+
+This method MUST callback <code>Update</code> method in native <code>ContractManagement</code> contract. It will update contract state if it's exectued successfully.
+
+Updating a smart contract MUST have <code>nefFile</code> or <code>manifest.json</code>, see reference of [NEP-15](https://github.com/neo-project/proposals/blob/master/nep-15.mediawiki) and [NEP-16](https://github.com/neo-project/proposals/blob/master/nep-16.mediawiki). If just update one of them at a time is OK, another SHOULD be set as null. It MUST be passed to ContractManagement.Update() in this method.
```

Comment:
I think maybe more details on the consequences here. I mean, the differences that can happen if updated in two calls.

Reply:
We can guide them to read the relevant NEP standards listed below.